transmission/transmission.go

```diff
@@ -47,6 +48,7 @@ type Honeycomb struct {
 	BlockOnResponse        bool          // whether to block or drop responses when the queue fills
 	UserAgentAddition      string
 	DisableGzipCompression bool // toggles gzip compression when sending batches of events
+	EnableZstdCompression  bool // toggles zstd compression when sending batches of events
```

Comment:
zstd and gzip should be exclusive options - it should be an error to ask the library to use both.